pkg/clients/dynatrace/server_error.go

```diff
@@ -0,0 +1,42 @@
+package dynatrace
+
+import (
+	"fmt"
+	"strings"
+)
+
+type serverErrorResponse struct {
+	ErrorMessage ServerError `json:"error"`
+}
+
+type ConstraintViolation struct {
+	Description       string `json:"description"`
+	Location          string `json:"location"`
+	Message           string `json:"message"`
+	ParameterLocation string `json:"parameterLocation"`
+	Path              string `json:"path"`
+}
+
+// ServerError represents an error returned from the server (e.g. authentication failure).
+type ServerError struct {
+	Message              string                `json:"message"`
+	ConstraintViolations []ConstraintViolation `json:"constraintViolations,omitempty"`
+	Code                 int                   `json:"code"`
+}
```

Comment:
is this correct?
[the docs](https://developer.dynatrace.com/plan/platform-services/error-handling/) show that it should be in a `details` section:
```
{
  "error": {
    "code": 400,
    "message": "Constraints violated.",
    "details": {
      "constraintViolations": [
        {
          "path": "detectionRules[0].filterConfig.pattern",
          "message": "may not be null",
          "parameterLocation": "PAYLOAD_BODY"
        }
      ]
    }
  }
}
```

the docs also say "These are commonly used fields in the `details` object", which reads to me that it is not always there, could that be a problem?

Reply:
let's make sure to not break handling of error responses!

@aorcholski  please check with cluster teams for the best way to handle errors if we are unsure of edge cases.

Reply:
did a double check, and IMO [the docs](https://developer.dynatrace.com/plan/platform-services/error-handling/) are just wrong or not meant for the tenant-apis, their is never a `details` section in the error response, even the examples in the swagger UI don't have them